lib/Doctrine/ORM/Mapping/ReflectionEnumProperty.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ReflectionProperty;
+use ReturnTypeWillChange;
+use UnitEnum;
+
+class ReflectionEnumProperty extends ReflectionProperty
+{
+    /** @var ReflectionProperty */
+    private $originalReflectionProperty;
+
+    /** @var class-string<UnitEnum> */
+    private $enumType;
+
+    public function __construct($originalReflectionProperty, string $enumType)
+    {
+        $this->originalReflectionProperty = $originalReflectionProperty;
+        $this->enumType                   = $enumType;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return mixed
+     */
+    #[ReturnTypeWillChange]
+    public function getValue($object = null)
+    {
+        $enum = $this->originalReflectionProperty->getValue($object);
+
+        if ($enum === null) {
+            return null;
+        }
+
+        return $enum->value;
+    }
+
+    #[ReturnTypeWillChange]
+    public function setValue($object, $value = null)
```

Comment:
```suggestion
    public function setValue($object, $value = null): void
```

There's no need to use that attribute for a brand-new class.

Reply:
You've marked this comment as resolved without applying the change or explaining why you don't want to apply it. 🙃

Reply:
@derrabus when you commented that it didn't extend `ReflectionProperty` yet, i thought doing that made the change impossible, but that was my mistake. Changed it now.